### ISSUE TITLE
salt/modules/disk.py list index out of range

### ISSUE THREAD

**Issue by hthu** [saltstack/salt#34805]
### Description of Issue/Question
When run `salt '*' disk.usage`, the above exception is thrown.
```
The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/site-packages/salt/minion.py", line 1200, in _thread_return
        return_data = func(*args, **kwargs)
      File "/usr/lib/python2.7/site-packages/salt/modules/disk.py", line 91, in usage
        while not comps[1].isdigit():
    IndexError: list index out of range
```

After a sanity check of my `df` command output, I sense that there is a "mal-formatted" line in the output, as:
"df: '/mnt/hgfs': Protocol error"

Even though this is a problem on my side, a better approach should be the code to deal with this kind of situation more elegantly?
Say, still giving output but state there is a error?

### Versions Report
Master:
```
Salt Version:
           Salt: 2016.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.10 (default, Oct 23 2015, 19:19:21)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4
            ZMQ: 4.1.5

System Versions:
           dist:
        machine: x86_64
        release: 15.5.0
         system: Darwin
        version: 10.11.5 x86_64
```

Minion:
```
  Salt: 2015.5.10
         Python: 2.7.11 (default, Jul  8 2016, 19:45:00)
         Jinja2: 2.8
       M2Crypto: 0.21.1
 msgpack-python: 0.4.7
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.7.0
           RAET: Not Installed
            ZMQ: 4.1.2
           Mako: Not Installed
        Tornado: Not Installed
        timelib: Not Installed
       dateutil: Not Installed
```

**Comment by rallytime** [comment]
@hthu Yeah, we definitely should be catching those errors and cleaning up those stacktraces more elegantly. Thanks for letting us know!

**Comment by thatch45** [comment]
@hthu it looks like this bug was fixed in 2016.3.1 here:
https://github.com/saltstack/salt/commit/8450df04831776419ae40989ec99a9216f915600
so I am going to close this out.

To fix it in your environment please upgrade your minion to 2016.3.1
